### DISCUSSION TITLE
Textbox z-index

### DISCUSSION THREAD

**Discussion by JamesGai** [hakimio/ngx-ui-tour#198] [Q&A] [answered]
Hi

I have changed the backdrop z-index because it will be override by the modal shadow root. How can I also change the z-index or there are some other approaches to bring the step textbox to the very front?
![image](https://github.com/user-attachments/assets/2f46ec3f-ab0f-4965-939d-b115032dccd4)

**Comment by JamesGai** [comment]
Or maybe the question is how to change the z-index of <ion-popver> element

**Reply by hakimio** [reply]
Active tour anchor element has `.touranchor--is-active` CSS class applied to it:
https://hakimio.github.io/ngx-ui-tour/md-menu/Misc#styling-active-anchor

You can use `popoverClass` config to apply custom CSS class to tour step:
https://hakimio.github.io/ngx-ui-tour/md-menu/API#step-config

**Reply by JamesGai** [reply]
I'm using Angular 15, Ionic 6, which results in ngx-ui-tour 1. The problem is steps in ngx-ui-tour 1 does not have `popoverClass` attribute. Also I have checked other discussions, modifying `.touranchor--is-active` can only change the highlighted element with anchorId not the underneath textbox right?

**Reply by hakimio** [reply]
Yes, you need `v2` to get `popoverClass` config.
To select the parent input you could use `input:has(.touranchor--is-active)` CSS selector.

**Reply by JamesGai** [reply]
Thanks a lot, I have tested `.touranchor--is-active` selector and everything works as expected. However, this selector only allows you to modify the styling of anchored elements not the tour guide textbox. 

How can I assign any stylings to the textbox and the corresponding elements on it?
![image](https://github.com/user-attachments/assets/2d2b62cb-3ba5-4bc0-905c-0008b8b1777b)

**Reply by hakimio** [reply, marked as answer]
`popoverClass` in `v2` or create custom tour step template.

**Reply by JamesGai** [reply]
I see, Thanks :)
